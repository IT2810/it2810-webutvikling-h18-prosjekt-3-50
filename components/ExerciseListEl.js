import React, { Component } from 'react'
import { View, Text, StyleSheet } from 'react-native'

import { Button, ListItem, Left, Right, Body } from 'native-base'

class ExerciseListEl extends Component {

  render () {
    return (
      <ListItem>
        <Left
          onPress={() => navigate('AddExercise')}
        >
          <Text> { this.props.exercise }  </Text>
        </Left>
        <Body
          onPress={() => navigate('AddExercise')}
        >
          <Text> 12 x 4 </Text>
          
        </Body>
        <Right>
          <Button
            transparent
            onPress={() => {}}
          >
            <Text> Delete </Text>
          </Button>
        </Right>
      </ListItem>
    )
  }
}

export default ExerciseListEl