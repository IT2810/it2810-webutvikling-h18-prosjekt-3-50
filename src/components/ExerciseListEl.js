import React, { Component } from 'react'
import { StyleSheet } from 'react-native'

import { Button, View, Text, ListItem, Left, Right, Body, Toast } from 'native-base'

class ExerciseListEl extends Component {

  constructor(props, context) {
    super(props, context)

    this.delete = this.delete.bind(this)
  }

  delete() {
    // TODO: delete from state
    Toast.show({
      type: 'warning',
      duration: 3000,
      text: 'Deleted exercise'
    })
  }

  render () {
    return (
      <ListItem>
        <Left
          onPress={() => navigate('AddExercise')}
        >
          <Text> { this.props.exercise }  </Text>
        </Left>
        <Body
          onPress={() => navigate('AddExercise')}
        >
          <Text> 12 x 4 </Text>
          
        </Body>
        <Right>
          <Button
            error
            onPress={this.delete}
          >
            <Text> X </Text>
          </Button>
        </Right>
      </ListItem>
    )
  }
}

export default ExerciseListEl