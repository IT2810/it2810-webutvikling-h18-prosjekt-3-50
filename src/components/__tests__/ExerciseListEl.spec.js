/**
 * @jest-environment jsdom
 */

import React from 'react'
import { shallow } from 'enzyme'
import ExerciseListEl from '../ExerciseListEl'
import toJson from 'enzyme-to-json';
import 'native-base'

import renderer from 'react-test-renderer';

describe('ExerciseListEl', () => {
  let wrapper
  const navigation = {navigate: jest.fn()}

  beforeEach(() => {
    wrapper = shallow(<ExerciseListEl navigation={navigation}/>)
  })

  it('renders correctly', () => {
    expect(toJson(wrapper.dive())).toMatchSnapshot()
  })

  // TODO: Can delete from list
})