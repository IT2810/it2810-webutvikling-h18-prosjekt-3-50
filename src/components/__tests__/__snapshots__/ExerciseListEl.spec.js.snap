// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExerciseListEl renders correctly 1`] = `
<ListItem
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderBottomWidth": 0.5,
      "borderColor": "#c9c9c9",
      "flexDirection": "row",
      "marginLeft": 16,
      "paddingBottom": 13,
      "paddingRight": 16,
      "paddingTop": 13,
    }
  }
>
  <Styled(Left)
    onPress={[Function]}
  >
    <Styled(Text)>
       
        
    </Styled(Text)>
  </Styled(Left)>
  <Styled(Body)
    onPress={[Function]}
  >
    <Styled(Text)>
       12 x 4 
    </Styled(Text)>
  </Styled(Body)>
  <Styled(Right)>
    <Styled(Button)
      danger={true}
      onPress={[Function]}
    >
      <Styled(Text)>
         X 
      </Styled(Text)>
    </Styled(Button)>
  </Styled(Right)>
</ListItem>
`;
